### PR TITLE
test: scripted-diff cleanups after generate* changes

### PR DESCRIPTION
**Pull request by maflcko**
Some cleanups after commit 94db963de501e4aba6e5d8150a01ceb85753dee1

### CODE REVIEW

File: test/functional/feature_blockfilterindex_prune.py [hunk 1]
```diff
@@ -26,9 +26,7 @@ def run_test(self):
         assert_greater_than(len(self.nodes[0].getblockfilter(self.nodes[0].getbestblockhash())['filter']), 0)
         # Mine two batches of blocks to avoid hitting NODE_NETWORK_LIMITED_MIN_BLOCKS disconnection
```
Comment (MarcoFalke) [inline test/functional/feature_blockfilterindex_prune.py:27]: This is wrong

Comment (MarcoFalke) [reply test/functional/feature_blockfilterindex_prune.py:27]: Ok, actually it doesn't matter: https://github.com/bitcoin/bitcoin/pull/23365#discussion_r745466739

File: test/functional/feature_backwards_compatibility.py [hunk 2]
```diff
@@ -66,8 +66,6 @@ def setup_nodes(self):
     def run_test(self):
         self.generatetoaddress(self.nodes[0], COINBASE_MATURITY + 1, self.nodes[0].getnewaddress())
 
-        self.sync_blocks()
```
Comment (benthecarman) [inline test/functional/feature_backwards_compatibility.py:69]: commit fa6c9964860cd154cf40aff74dfa038aa6006de0 says to remove `sync_all` but you are removing `sync_blocks` as well

**Comment by fanquake** [comment]
Concept ACK

File: test/functional/feature_backwards_compatibility.py (hunk 2 above)
Comment (MarcoFalke) [reply test/functional/feature_backwards_compatibility.py:69]: Thanks, adjusted the commit subject

**Comment by theStack** [comment]
Concept ACK

**Review by fanquake — APPROVED** [review]
ACK fac23c211407a77af82bb1491c48c8d37022c8b3

**Pull request merged**